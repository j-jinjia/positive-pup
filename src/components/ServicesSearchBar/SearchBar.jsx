import "./SearchBar.scss";
import searchIcon from "../../assets/svgs/searchBarIcon.svg";

const ServicesSearchBar = ({ searchTerm, handleInput }) => {
  return (
    <div className="search-box">
      <label htmlFor="search-box">
      <input
        className="search-box__input"
        type="text"
        value={searchTerm}
        onChange={handleInput}
      />
      </label>
     
      <img
        className="search-box__input-icon"
        src={searchIcon}
        alt="search icon"
      />
    </div>
  );
};

export default ServicesSearchBar;
