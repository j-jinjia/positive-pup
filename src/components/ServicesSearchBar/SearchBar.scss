@use "../../styles/variables" as *;


.search-box {
  position: relative;

  &__input {
    width: 15rem;
    height: 2.5rem;
    position: relative;
    border-radius: 8px;
    background-color: $colour-primary-light-searchbar;
    border: none;

    &-icon {
      position: absolute;
      right: 10px;
      top: 10px;
    }
  }
}
