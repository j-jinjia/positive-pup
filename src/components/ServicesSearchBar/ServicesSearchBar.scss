.search-box {
  position: relative;

  &__input {
    width: 242px;
    height: 40px;
    position: relative;
    border-radius: 8px;
    background-color: rgba(55, 175, 162, 0.04);
    border: none;

    &--icon {
      position: absolute;
      right: 10px;
      top: 10px;
    }
  }
}
