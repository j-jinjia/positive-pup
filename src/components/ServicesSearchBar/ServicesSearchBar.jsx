import "./ServicesSearchBar.scss";
import searchIcon from "../../assets/svgs/searchBarIcon.svg";

const ServicesSearchBar = ({ searchTerm, handleInput }) => {
  return (
    <div className="search-box">
      <input
        className="search-box__input"
        type="text"
        value={searchTerm}
        onChange={handleInput}
      />
      <img
        className="search-box__input--icon"
        src={searchIcon}
        alt="search icon"
      />
    </div>
  );
};

export default ServicesSearchBar;
