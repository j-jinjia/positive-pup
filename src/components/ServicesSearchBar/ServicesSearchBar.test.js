import { customRender } from "../../utils/testUtils";
import ServicesSearchBar from "./ServicesSearchBar";

it("should match ServicesSearchBar component snapshot", () => {
  const { container } = customRender(<ServicesSearchBar/>);

  expect(container).toMatchSnapshot();
});