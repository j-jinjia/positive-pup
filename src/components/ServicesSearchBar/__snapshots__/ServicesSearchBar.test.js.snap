// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match ServicesSearchBar component snapshot 1`] = `
<div>
  <div
    class="search-box"
  >
    <input
      class="search-box__input"
      type="text"
      value=""
    />
    <img
      alt="search icon"
      class="search-box__input--icon"
      src="searchBarIcon.svg"
    />
  </div>
</div>
`;
